pilot/pkg/proxy/envoy/v2/discovery.go

```diff
@@ -169,7 +172,12 @@ func (s *DiscoveryServer) ClearCacheFunc() func() {
 		// saved.
 		s.env.PushStatus = model.NewStatus()
 
-		AdsPushAll(s)
+		// TODO: propagate K8S version and use it instead
+		versionMutex.Lock()
+		version = time.Now().Format(time.RFC3339)
+		versionMutex.Unlock()
```

Comment:
what is the impact of sending a different version with each push? Lets make sure it doesn't cause unnecessary listener reloads which in turn causes long lived TCP connections to break. We just fixed that issue

Reply:
Well - that's what I'm fixing, update versionInfo only on change.

Reply:
But to answer your question - it doesn't seem that it was causing reloads.

Reply:
This block moved from AdsPushAll ( called from here and from periodic ).